core/vector.h

```diff
@@ -34,7 +34,7 @@
 /**
  * @class Vector
  * @author Juan Linietsky
- * Vector container. Regular Vector Container. Use with care and for smaller arrays when possible. Use Vector for large arrays.
+ * Vector container. Regular Vector Container.
```

Comment:
Could you change it to:
```
Vector container that uses a Copy on Write mechanism for passing data around efficiently.
```

Reply:
(Note: I cancelled the CI builds as a change in `vector.h` means a 50 min build stalling every other PR, so I prefer to keep it only for the final approved change ;))